hardware_interface/include/hardware_interface/robot_hw.h

```diff
@@ -101,8 +101,16 @@ class RobotHW : public InterfaceManager
    * Check (in non-realtime) if given controllers could be started and stopped from the current state of the RobotHW
    * with regard to necessary hardware interface switches. Start and stop list are disjoint.
    * This is just a check, the actual switch is done in doSwitch()
+   * @deprecated: Implement prepareSwitch() instead
    */
-  virtual bool canSwitch(const std::list<ControllerInfo>& /*start_list*/, const std::list<ControllerInfo>& /*stop_list*/) const { return true; }
+  virtual bool canSwitch(const std::list<ControllerInfo>& /*start_list*/, const std::list<ControllerInfo>& /*stop_list*/) { return true; }
```

Comment:
Update:
g++ will print plenty of deprecation warnings, especially for every controller because `controller_base.h` includes `robot_hw.h`.
clang++ is less noisy, it warns only if `canSwitch` is accessed directly.

So I would suggest to not add `ROS_DEPRECATED` is this special case.